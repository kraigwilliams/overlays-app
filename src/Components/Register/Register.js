import React, { Component } from 'react'
import { Button, Input, Required } from '../Utils/Utils'
import AuthApiService from '../../Services/auth-api-service'

/*
RegistrationForm renders a form for a new user to register.
*/
export default class RegistrationForm extends Component {
  static defaultProps = {
    onRegistrationSuccess: () => {}
  }

  state = { error: null }

componentDidMount(){
  document.title= "Create Account - Overlays"
}


  handleSubmit = event => {
    event.preventDefault()
    const { user_name, user_password } = event.target

    this.setState({ error: null })
    AuthApiService.postUser({
      user_name: user_name.value,
      user_password: user_password.value,
     
    })
      .then(user => {
      
        user_name.value = ''
        user_password.value = ''
        this.props.onRegistrationSuccess()
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }

  render() {
    const { error } = this.state
    return (
      <form
        className='RegistrationForm'
        onSubmit={this.handleSubmit}
      >
        <div role='alert'>
          {error && <p className='red'>{error}</p>}
        </div>
       
        <div className='user_name'>
          <label htmlFor='RegistrationForm__user_name'>
           Create User name <Required />
          </label>
          <Input
            name='user_name'
            type='text'
            required
            id='RegistrationForm__user_name'>
          </Input>
        </div>
        <div className='password'>
          <label htmlFor='RegistrationForm__password'>
           Set Password <Required />
          </label>
          <Input
            name='user_password'
            type='password'
            required
            id='RegistrationForm__password'>
          </Input>
        </div>
       
        <Button type='submit'>
          Register
        </Button>
      </form>
    )
  }
}