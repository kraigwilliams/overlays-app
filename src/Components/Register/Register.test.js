import React from 'react';
//import { render } from '@testing-library/react';
import Register from './Register'
import {BrowserRouter} from 'react-router-dom'
import ReactDOM from  'react-dom';
import renderer from 'react-test-renderer';

describe('<Register/>',()=>{
  it ('is able to render',()=>{
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><Register/></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<BrowserRouter><Register/></BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });

  
})