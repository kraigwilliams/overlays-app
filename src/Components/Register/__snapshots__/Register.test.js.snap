// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register/> renders the UI as expected 1`] = `
<form
  className="RegistrationForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="user_name"
  >
    <label
      htmlFor="RegistrationForm__user_name"
    >
      Create User name 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input "
      id="RegistrationForm__user_name"
      name="user_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="RegistrationForm__password"
    >
      Set Password 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input "
      id="RegistrationForm__password"
      name="user_password"
      required={true}
      type="password"
    />
  </div>
  <button
    className="Button "
    type="submit"
  >
    Register
  </button>
</form>
`;
